cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp

```diff
@@ -79,6 +81,214 @@ struct vertex_similarity_ops {
+        const descriptor_base &descriptor,
+        const dal::preview::detail::topology<std::int32_t> &data,
+        void *result_ptr);
+    virtual ~backend_default() {}
+};
+
+template <typename Policy, typename Float, class Method, typename Topology>
+dal::detail::pimpl<backend_base<Policy, Topology>> get_backend(const descriptor_base &desc,
+                                                               const Topology &data) {
+    return dal::detail::pimpl<backend_base<Policy, Topology>>(
+        new backend_default<Policy, float, method::by_default, Topology>);
+}
+
+inline std::size_t get_number_elements_in_block(const std::int32_t &row_range_begin,
+                                                const std::int32_t &row_range_end,
+                                                const std::int32_t &column_range_begin,
+                                                const std::int32_t &column_range_end) {
+    ONEDAL_ASSERT(row_range_end >= row_range_begin, "Negative interval found");
+    const std::size_t row_count = row_range_end - row_range_begin;
+    ONEDAL_ASSERT(column_range_end >= column_range_begin, "Negative interval found");
+    const std::size_t column_count = column_range_end - column_range_begin;
+    // compute the number of the vertex pairs in the block of the graph
+    const std::size_t vertex_pairs_count = row_count * column_count;
+    ONEDAL_ASSERT(vertex_pairs_count / row_count == column_count,
+                  "Overflow found in multiplication of two values");
+    return vertex_pairs_count;
+}
+
+template <typename Float, typename Index>
+inline std::size_t get_max_block_size(const std::int64_t &vertex_pairs_count) {
+    const std::size_t vertex_pair_element_count = 2; // 2 elements in the vertex pair
+    const std::size_t jaccard_coeff_element_count = 1; // 1 Jaccard coeff for the vertex pair
+
+    const std::size_t vertex_pair_size = vertex_pair_element_count * sizeof(Index); // size in bytes
+    const std::size_t jaccard_coeff_size =
+        jaccard_coeff_element_count * sizeof(Float); // size in bytes
+    const std::size_t element_result_size = vertex_pair_size + jaccard_coeff_size;
+
+    const std::size_t block_result_size = element_result_size * vertex_pairs_count;
+    ONEDAL_ASSERT(block_result_size / vertex_pairs_count == element_result_size,
+                  "Overflow found in multiplication of two values");
+    return block_result_size;
+}
+
+template <typename Policy, typename Float, class Method, typename Graph>
+vertex_similarity_result vertex_similarity_ops_dispatcher<Policy, Float, Method, Graph>::operator()(
+    const Policy &policy,
+    const descriptor_base &desc,
+    vertex_similarity_input<Graph> &input) const {
+    const auto &csr_topology =
+        dal::preview::detail::csr_topology_builder<Graph>()(input.get_graph());
+    const auto row_begin =
+        dal::detail::integral_cast<vertex_type<Graph>>(desc.get_row_range_begin());
```

Comment:
probably you can use function alias
```
template<typename Graph>
const auto integral_cast_func = dal::detail::integral_cast<vertex_type<Graph>>;
```
and use as
`integral_cast_func(desc.get_row_range_begin());`

Reply:
removed